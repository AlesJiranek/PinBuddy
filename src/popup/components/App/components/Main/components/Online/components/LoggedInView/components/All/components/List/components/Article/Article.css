@keyframes articleLoading {
  to {
    transform: rotate(360deg);
  }
}

.article {
  display: block;
  text-decoration: none;
  position: relative;
}

.article--private {
  background-color: var(--color-gray-light);
}

.article__loading {
  animation-name: fadeIn;
  animation-duration: var(--animation-duration);
  animation-fill-mode: backwards;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
}

.article__loading-icon {
  border-radius: 50%;
  width: calc(var(--lh) - 2px);
  height: calc(var(--lh) - 2px);
  border: 2px solid var(--color-gray-light);
  border-top-color: var(--color-primary);
  animation-name: articleLoading;
  animation-duration: calc(var(--animation-duration) * 4);
  animation-iteration-count: infinite;
  animation-timing-function: cubic-bezier(.3, .1, .3, .85);
}

.article__url {
  text-decoration: none;
}

.article__url:hover {
  text-decoration: underline;
}

.article__url--unread {
  color: var(--color-secondary);
}
.article__meta {
  font-size: .8rem;
  color: var(--color-gray);
}

.article__metatop {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.article__metabottom {
  display: flex;
}

.article__metainfo::before {
  content: '|';
  margin: 0 calc(var(--lh) / 4);
}

.article__metainfo:first-child::before {
  display: none;
}

.article__metainfo--tags {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.article__cancel,
.article__delete {
  cursor: pointer;
}

.article__cancel:hover {
  color: var(--color-primary);
}

.article__separator {
  margin: calc(var(--lh) / 4);
}

.article__delete:hover {
  color: var(--color-error);
}
