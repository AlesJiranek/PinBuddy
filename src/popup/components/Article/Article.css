.article {
  display: block;
  text-decoration: none;
}

.article--private {
  background-color: var(--color-gray-light);
}

.article__url {
  text-decoration: none;
}

.article__url--unread {
  color: var(--color-secondary);
}

.article__url:hover {
  text-decoration: underline;
}

.article__meta {
  font-size: .8rem;
  color: var(--color-gray);
  display: flex;
}

.article__info {
  margin-right: calc(var(--lh) / 3);
}

.article__info:last-child {
  margin-right: 0;
}

.article__info--tags {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.article__info--edit:hover,
.article__info--delete:hover {
  cursor: pointer;
}

.article__info--edit:hover {
  color: var(--color-primary);
}

.article__info--delete:hover {
  color: var(--color-error);
}
