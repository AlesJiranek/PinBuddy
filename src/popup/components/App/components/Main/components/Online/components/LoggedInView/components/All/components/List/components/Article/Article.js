import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { postsDelete } from 'redux-popup/actions/posts';
import './Article.css';

export class Article extends Component {
  state={
    deleteActive: false,
    deleteConfirmed: false,
    focused: false,
  }

  linkRef = React.createRef();


  componentDidMount() {
    addEventListener('keydown', this.handleKeyDownEvent);
  }

  componentWillUnmount() {
    removeEventListener('keydown', this.handleKeyDownEvent);
  }

  render() {
    const {
      privatePost,
      unread,
      href,
      description,
      time,
      tags,
    } = this.props;

    const timeFormated = time.substring(0, 10).replace(/-/g, '.');
    const tagsFormated = tags.split(' ').map(tag => `#${tag}`).join(' ');
    const { deleteActive, deleteConfirmed } = this.state;

    return (
      <article className={`article ${privatePost ? 'article--private' : ''}`}>
        {
          deleteConfirmed && (
            <div className="article__loading">
              <div className="article__loading-icon" />
            </div>
          )
        }
        <a
          tabIndex={deleteConfirmed ? '-1' : '0'}
          className={`article__url ${unread ? 'article__url--unread' : ''}`}
          href={decodeURIComponent(href)}
          title={description}
          rel="noopener noreferrer"
          target="_blank"
          onFocus={this.handleOnFocusLink}
          onBlur={this.handleOnBlursLink}
          ref={this.linkRef}
        >
          {description}
        </a>
        <div className="article__meta">
          <div className="article__info">{timeFormated}</div>
          {tags ? <div className="article__info article__info--tags">{tagsFormated}</div> : ''}
          <div className="article__info">
            {deleteActive && (
              <>
                <span className="article__info--cancel" onClick={this.handleCancelClick}>{chrome.i18n.getMessage('popupArticleButtonCancel')}</span>
                <span className="article__info--separator">/</span>
              </>
            )}
            {!deleteConfirmed && (
              <span className="article__info--delete" onClick={this.handleDeleteClick}>{chrome.i18n.getMessage('popupArticleButtonDelete')}</span>
            )}
          </div>
        </div>
      </article>
    );
  }

  handleDeleteClick = () => {
    const { href, description, extended, postsDelete } = this.props;
    if (this.state.deleteActive) {
      this.setState({
        deleteConfirmed: true,
      });
      postsDelete(href, description, extended);
    }

    this.setState(state => {
      return {
        deleteActive: !state.deleteActive,
      };
    });
  }

  handleCancelClick = () => {
    this.setState(state => {
      return {
        deleteActive: !state.deleteActive,
      };
    });
  }

  handleOnFocusLink = () => {
    this.setState({
      focused: true,
    });
  }

  handleOnBlursLink = () => {
    this.setState({
      focused: false,
    });
  }

  handleKeyDownEvent = e => {
    const { href, description, extended, postsDelete } = this.props;

    if ((e.ctrlKey || e.metaKey) && e.keyCode === 8 && this.state.focused) {
      this.linkRef.current.blur();
      this.setState(state => ({
        deleteActive: true,
        deleteConfirmed: true,
      }), () => postsDelete(href, description, extended));
    }
    else if (e.keyCode === 13 && (e.ctrlKey || e.metaKey) && this.state.focused) {
      e.preventDefault();
      chrome.tabs.create({
        url: this.props.href,
        active: false,
      });
    }
  }

}

Article.propTypes = {
  privatePost: PropTypes.bool.isRequired,
  unread: PropTypes.bool.isRequired,
  href: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  extended: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  postsDelete: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  postsDelete: (href, title, description) => dispatch(postsDelete(href, title, description)),
});

export default connect(null, mapDispatchToProps)(Article);
