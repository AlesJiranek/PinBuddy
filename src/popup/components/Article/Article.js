import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { postsDelete } from '../../actions/posts';
import './Article.css';

class Article extends Component {
  state={
    deleteActive: false,
  }

  render() {
    const {
      privatePost,
      unread,
      href,
      description,
      time,
      tags,
    } = this.props;

    const timeFormated = time.substring(0, 10).replace(/-/g, '.');
    const tagsFormated = tags.split(' ').map(tag => `#${tag}`).join(' ');

    return (
      <article className={`artile ${privatePost ? 'article--private' : ''}`}>
        <a
          className={`article__url ${unread ? 'article__url--unread' : ''}`}
          href={href}
          rel="noopener noreferrer"
          target="_blank"
        >
          {description}
        </a>
        <div className="article__meta">
          <div className="article__info">{timeFormated}</div>
          {tags ? <div className="article__info article__info--tags">{tagsFormated}</div> : ''}
          <div className="article__info article__info--edit">Edit</div>
          <div className="article__info">
            {this.state.deleteActive && (
              <>
                <span className="article__info--cancel" onClick={this.handleCancelClick}>Cancel</span>
                <span className="article__info--separator">/</span>
              </>
            )}
            <span className="article__info--delete" onClick={this.handleDeleteClick}>Delete</span>
          </div>
        </div>
      </article>
    );
  }

  handleDeleteClick = () => {
    if (this.state.deleteActive) {
      this.props.postsDelete(this.props.href);
    }

    this.setState(state => {
      return {
        deleteActive: !state.deleteActive,
      };
    });
  }

  handleCancelClick = () => {
    this.setState(state => {
      return {
        deleteActive: !state.deleteActive,
      };
    });
  }
}


Article.propTypes = {
  privatePost: PropTypes.bool.isRequired,
  unread: PropTypes.bool.isRequired,
  href: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  postsDelete: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => {
  return {
    postsDelete: href => dispatch(postsDelete(href)),
  };
};

export default connect(null, mapDispatchToProps)(Article);
