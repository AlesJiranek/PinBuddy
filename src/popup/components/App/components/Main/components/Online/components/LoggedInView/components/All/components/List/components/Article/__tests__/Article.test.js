import React from 'react';
import { shallow } from 'enzyme';

import { Article } from '../Article';

describe('<Article />', () => {

  const defaultProps = {
    privatePost: true,
    unread: true,
    href: '',
    description: '',
    extended: '',
    time: '',
    tags: '',
    postsDelete: jest.fn(),
  };

  it('should render correctly', () => {
    const tree = shallow(
      <Article
        {...defaultProps}
      />
    );
    expect(tree).toMatchSnapshot();
  });

});
