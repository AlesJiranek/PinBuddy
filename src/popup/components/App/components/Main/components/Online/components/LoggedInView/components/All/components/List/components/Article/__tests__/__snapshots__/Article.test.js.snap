// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    description=""
    extended=""
    href=""
    postsDelete={[MockFunction]}
    privatePost={true}
    tags=""
    time=""
    unread={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <a
          className="article__url article__url--unread"
          href=""
          onBlur={[Function]}
          onFocus={[Function]}
          rel="noopener noreferrer"
          tabIndex="0"
          target="_blank"
          title=""
        >
          
        </a>,
        <div
          className="article__meta"
        >
          <div
            className="article__info"
          >
            
          </div>
          
          <div
            className="article__info"
          >
            <span
              className="article__info--delete"
              onClick={[Function]}
            >
              popupArticleButtonDelete
            </span>
          </div>
        </div>,
      ],
      "className": "article article--private",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "article__url article__url--unread",
          "href": "",
          "onBlur": [Function],
          "onFocus": [Function],
          "rel": "noopener noreferrer",
          "tabIndex": "0",
          "target": "_blank",
          "title": "",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": "",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="article__info"
            >
              
            </div>,
            "",
            <div
              className="article__info"
            >
              <span
                className="article__info--delete"
                onClick={[Function]}
              >
                popupArticleButtonDelete
              </span>
            </div>,
          ],
          "className": "article__meta",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "article__info",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                <span
                  className="article__info--delete"
                  onClick={[Function]}
                >
                  popupArticleButtonDelete
                </span>,
              ],
              "className": "article__info",
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "popupArticleButtonDelete",
                  "className": "article__info--delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "popupArticleButtonDelete",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <a
            className="article__url article__url--unread"
            href=""
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener noreferrer"
            tabIndex="0"
            target="_blank"
            title=""
          >
            
          </a>,
          <div
            className="article__meta"
          >
            <div
              className="article__info"
            >
              
            </div>
            
            <div
              className="article__info"
            >
              <span
                className="article__info--delete"
                onClick={[Function]}
              >
                popupArticleButtonDelete
              </span>
            </div>
          </div>,
        ],
        "className": "article article--private",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "article__url article__url--unread",
            "href": "",
            "onBlur": [Function],
            "onFocus": [Function],
            "rel": "noopener noreferrer",
            "tabIndex": "0",
            "target": "_blank",
            "title": "",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": "",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="article__info"
              >
                
              </div>,
              "",
              <div
                className="article__info"
              >
                <span
                  className="article__info--delete"
                  onClick={[Function]}
                >
                  popupArticleButtonDelete
                </span>
              </div>,
            ],
            "className": "article__meta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "article__info",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  false,
                  <span
                    className="article__info--delete"
                    onClick={[Function]}
                  >
                    popupArticleButtonDelete
                  </span>,
                ],
                "className": "article__info",
              },
              "ref": null,
              "rendered": Array [
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "popupArticleButtonDelete",
                    "className": "article__info--delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "popupArticleButtonDelete",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
